File: srm171/RPG.java
import java.lang.*;

public class RPG {

	public int[] dieRolls(String[] dice) {
		int[] mma = new int[3];
		for (int i = 0; i < dice.length; i++) {
			int tmp1 = Integer.parseInt(dice[i].substring(0, dice[i].indexOf('d')));
			int tmp2 = Integer.parseInt(dice[i].substring((dice[i].indexOf('d')+1), dice[i].length()));			
			mma[0] += tmp1;
			mma[1] += (tmp1 * tmp2);
		}
		double n = (mma[0] + mma[1]) / 2;
		mma[2] = (int)n;
		return mma;
	}
}

File: srm171/rpg.py
class RPG:
	def dieRolls(self, dice):
		mma = [0, 0, 0]
		for i in range(len(dice)):
			mma[0] += int(dice[i][:dice[i].index('d')])
			mma[1] += int(dice[i][dice[i].index('d')+1:]) * int(dice[i][:dice[i].index('d')])
		
		mma[2] = (mma[0] + mma[1]) / 2
		
		return mma